kore/src/Kore/ASTVerifier/SentenceVerifier.hs

```diff
@@ -358,12 +384,100 @@ verifyAxiomSentence sentence =
                 (freeVariables sentence)
                 (sentenceAxiomAttributes sentence)
         State.modify $ addAxiom verified attrs
+        either
+            (\case
+                RequiresError _ ->
+                    koreFailWithLocations
+                    [sentenceAxiomPattern verified]
+                        "RequiresError thrown during equation check"
+                ArgumentError _ ->
+                    koreFailWithLocations
+                    [sentenceAxiomPattern verified]
+                        "ArgumentError thrown during equation check"
+                AntiLeftError _ ->
+                    koreFailWithLocations
+                    [sentenceAxiomPattern verified]
+                        "AntiLeftError thrown during equation check"
+                EnsuresError _ ->
+                    koreFailWithLocations
+                    [sentenceAxiomPattern verified]
+                        "EnsuresError thrown during equation check"
+                NotEquation _ -> return ()
+                FunctionalAxiom -> return ()
+                ConstructorAxiom -> return ()
+                SubsortAxiom -> return ()
+            )
+            (\ Equation {left, argument, attributes} ->
+                when (isNotSimplification attributes)
+                    $ checkLHS left >> checkArg argument
+            )
+            $ fromSentenceAxiom (attrs, verified)
   where
     addAxiom verified attrs =
         Lens.over
             (field @"indexedModuleAxioms")
             ((attrs, verified) :)
 
+    isNotSimplification attributes =
+        Attribute.simplification attributes ==
+        NotSimplification
+
+    checkLHS termLike =
+        if isNonconstructorFunctionSymbol termLike
+        then failOnJust
+            "LHS of NotSimplification axiom contains non-variable:"
+            $ asum $ getNotVar <$> termLikeF
+        else koreFail $ show $ Pretty.vsep
+            [ "Head of LHS of Notsimplification axiom is not a non-constructor function symbol. This is the LHS:"
+            , unparse termLike
+            ]
+      where
+        _ :< termLikeF = Recursive.project termLike
+
+        isNonconstructorFunctionSymbol term
+          | App_ sym _ <- term =
+            Symbol.isFunction sym
+            && not (Symbol.isConstructorLike sym)
+          | otherwise = False
+
+        getNotVar (Var_ _) = Nothing
+        getNotVar term = Just term
+
+    checkArg Nothing = return ()
+    checkArg (Just arg) =
+        traverse_
+            ( failOnJust "Found invalid subterm of Axiom equation argument:"
+            . checkArgIn
+            )
+        $ Predicate.getMultiAndPredicate arg
+      where
+        checkArgIn (PredicateIn (Var_ _) term) = findSubterm
+            isBadArgSubterm
+            term
+        checkArgIn badArg = Just $ Predicate.fromPredicate_ badArg
+
+        isBadArgSubterm (App_ sym _) = not (Symbol.isConstructorLike sym)
+        isBadArgSubterm (DV_ _ _) = False
```

Comment:
You need to add a case to accept `StringLiteral` here, because the `DV_` will always have one for a child.